doc/man3/OPENSSL_LH_COMPFUNC.pod

```diff
@@ -177,19 +177,10 @@ lh_TYPE_free(), lh_TYPE_doall() and lh_TYPE_doall_arg() return no values.
 
 =head1 NOTE
 
-The various LHASH macros and callback types exist to make it possible
-to write type-checked code without resorting to function-prototype
-casting - an evil that makes application code much harder to
-audit/verify and also opens the window of opportunity for stack
-corruption and other hard-to-find bugs.  It also, apparently, violates
-ANSI-C.
-
-The LHASH code is not thread safe. All updating operations must be
-performed under a write lock. All retrieve operations should be performed
-under a read lock, I<unless> accurate usage statistics are desired.
-In which case, a write lock should be used for retrieve operations
-as well.  For output of the usage statistics, using the functions from
-L<OPENSSL_LH_stats(3)>, a read lock suffices.
+The LHASH code is not thread safe; all operations in a multi-threaded
+program should be done under a lock. If lh_TYPE_error() is called, then
+the lock should be held across both the original operation (e.g.,
+lh_TYPE_insert()) and the call to fetch teh error status.
```

Comment:
Please specify what kind of lock. There are read locks and there are write locks.

Reply:
No, it doesn't matter.  As long as the application locks, it is okay.

Reply:
I think you meant write locks. (Though see my note below. I'm not sure you want to do that.) Read locks still allow multiple threads to access the LHASH at the same time. They just interact appropriately with threads taking write locks.

In fact, the offending calls are already wrapped under read locks. TSan is (correctly) still flagging that because the stats are being updated in a way that read locks do not protect against.

PS: s/teh/the/